text/0156-columnar-encoding.md

```diff
@@ -0,0 +1,1417 @@
+  related metrics sharing the same attributes and timestamp) into a collection of univariate time-series resulting in a
+  large amount of duplication and additional overhead covering the entire chain from exporters to backends.
+* **Provide more advanced and efficient telemetry data processing capabilities**. Increasing data volume, cost
+  efficiency, and data minimization require additional data processing capabilities such as data projection,
+  aggregation, and filtering.
+
+These improvements not only address the aforementioned needs but also answer the [open questions](https://github.com/open-telemetry/oteps/blob/main/text/0035-opentelemetry-protocol.md#open-questions)
+cited in OTEP 035 (i.e. cpu usage, memory pressure, compression optimization).
+
+**It is important to understand that this proposal is complementary to the existing protocol. The row-oriented version
+is still suitable for some scenarios. Telemetry sources that generate a small amount of telemetry data should continue
+to use it. On the other side of the spectrum, sources and collectors generating or aggregating a large amount of
+telemetry data will benefit from adopting this extension to optimize the resources involved in the transfer and
+processing of this data. This adoption can be done incrementally.**
+
+Before detailing the specifications of the OTLP Arrow protocol, the following two sections present: 1) a validation of
+the value of a columnar approach based on a set of benchmarks, 2) a discussion of the value of using Apache Arrow as a
+basis for columnar support in OTLP.
+
+### Validation
+
+A series of tests were conducted to compare compression ratios between OTLP and a columnar version of OTLP called OTLP
+Arrow. The two key results are:
+
+* For multivariate time series, OTLP Arrow is **4 times better in terms of bandwidth reduction while improving by
+  3 the speed** of creation + serialization + compression + decompression + deserialization.
+* For logs and traces, OTLP Arrow is **2 times better in terms of bandwidth reduction** while having only a very slight
+  speed drop for the creation + serialization + compression + decompression + deserialization steps.
+
+![Summary](img/0156_summary.png)
+
+Bandwidth gains between *OTLP* and *OTLP Arrow* in a multivariate time series context are represented in the left column.
+Similarly, logs are represented in the right column. For both protocols, the baseline is the size of the uncompressed
+OTLP messages. The reduction factor is the ratio between this baseline and the compressed message size for each
+protocol.
+
+The following two stacked bar graphs compare side-by-side the distribution of time spent for each step and for each
+version of the protocol. The 3x speed-up for the multivariate metrics is visible in the left column. The two protocols
+are on par for logs, as shown in the right column.
+
+![Summary of the time spent](img/0156_summary_time_spent.png)
```

Comment:
I agree, there is a mix between the benefits/conclusions of phase 1 and 2 of this project (phase 1 did not exist at the beginning of this project). Let me clarify that. 

In phase 1, the goal is to improve bandwidth and network costs (especially internet). 

Exporter steps:
* **Conversion to OTLP Arrow** => x1.5 to x3 slower than the creation of the equivalent OTLP message **[1]**.
* **Arrow serialization** => No serialization cost (zero-copy).
* **Protobuf serialization** => Negligible because `OtlpArrowRecords` is composed of very few fields.
* **Compression** => faster because smaller payload.

Receiver steps:
* **Decompression** => faster because smaller payload.
* **Protobuf deserialization** => Negligible because `OtlpArrowRecords` is composed of very few fields.
* **Arrow deserialization** => No deserialization cost (zero-copy).
* **Conversion to OTLP msg** => close to the time spent to deserialize/create the equivalent OTLP message in the standard receiver **[2]**.

**[1]**: these steps will not exist with the OTLP Arrow Client SDK and in phase 2 of the collector .
**[2]**: this step doesn't exist in phase 2 of the collector.

**In conclusion, phase 1 improves bandwidth and network costs at the expense of a slowdown on the exporter side. Phase 2 keeps the benefits of phase 1 but also improves the receiver/exporter speed and the data processing speed. I will add elements in the OTEP to improve the overall message.**

Reply:
I am thinking of integrating this representation of benefits per phase in the OTEP. What do you think?

![OTEL - BenefitsPhases](https://user-images.githubusercontent.com/657994/213897214-33adb55f-730c-4c00-b318-14705e65e000.png)

Reply:
@tigrannajaryan I have updated the validation section to better represent the time spent in all the different steps (including the OTLP to/from OTLP Arrow conversion). As you can see in phase 1, we get a significant improvement in compression ratio but at the cost of increased CPU and memory. In phase 2, OTLP Arrow is both faster end-to-end and offers a similar compression ratio as phase 1. These new charts no longer mix phase 1 and phase 2 and I think it is definitely less misleading. Thanks for this comment.